### DISCUSSION TITLE
What is the style for argument values (blue in bright color scheme -a=THIS_STYLE)

### DISCUSSION THREAD

**Discussion by eugenesvk** [pacak/bpaf#399] [Q&A] [answered]
I think only 4 are available
```rs
  let mut d = Doc::default();
  d.emphasis("emphasis ");
  d.literal("literal ");
  d.invalid("invalid ");
  d.text("text ");
```
But this is yellow, green, red, white, nothing matches the blue you get for argument value suggestions, which would be helpful if you enumerate such values in the comments

<img width="352" alt="color" src="https://github.com/user-attachments/assets/878b9bb2-841a-4718-a553-ec1d543177c6">

**Comment by pacak** [comment]
I'm pretty sure that's `literal`, this is the code used: https://github.com/pacak/bpaf/blob/88e2f4d116e629b218c669d6786053c2a78c3a26/src/meta_help.rs#L503-L521

**Reply by eugenesvk** [reply]
where is the `=` char in this code?
<img width="352" alt="color" src="https://github.com/user-attachments/assets/878b9bb2-841a-4718-a553-ec1d543177c6">



I see `=` here, could this be it?
```rs
        } => {
            buf.token(Token::BlockStart(Block::ItemTerm));
            write_shortlong(buf, *name);
            buf.write_str("=", Style::Text);
            buf.metavar(*metavar);
            buf.token(Token::BlockEnd(Block::ItemTerm));
```

**Reply by eugenesvk** [reply]
by the way, is it possible to reformat the `.argument::<String>("fDMETHOD")` text so that I could fit help here like `d`|`f`|`k` where | symbols wouldn't be colored, but actual argument options `d` etc would?

**Reply by pacak** [reply, marked as answer]
> I see `=` here, could this be it?

Yeah. `write_shortlong` deals with `--via` part, `=` goes as plain text, `<fDMETHOD>` is  gets rendered with `.metavar`.

At the moment `.argument` method takes `&'static str` as metavar so you can't customize it with `Doc`, but you can still put some ANSI sequence inside to override it to your liking.

**Reply by eugenesvk** [reply]
Ha, if I add ANSI stuff to replace `"fDMETHOD"`, it seems the width calculation breaks? 
`format!("{}|{}|{}","d".blue().bold(),"f".blue().bold(),"os".blue().bold());`
leads to this misalignment

<img width="108" alt="bpaf" src="https://github.com/user-attachments/assets/d6d73827-23ad-45ad-a508-949919baa9b9">

(note that the closing `>` isn't styled anymore)

**Reply by eugenesvk** [reply]
I think direct ANSI colors also negatively affect the width calculations for the comments as now I get word wrapping much earlier
